src/handlers/gestures/simpleGestures.ts

```diff
@@ -0,0 +1,381 @@
+import { Gesture } from './gesture';
+import { GestureBuilder, BuiltGesture } from './gestureBuilder';
+import { Directions } from '../../Directions';
+import { baseGestureHandlerWithMonitorProps } from '../gestureHandlerCommon';
+import { getNextHandlerTag } from '../handlersRegistry';
+import { tapGestureHandlerProps } from '../TapGestureHandler';
+import {
+  panGestureHandlerProps,
+  panGestureHandlerCustomNativeProps,
+  managePanProps,
+} from '../PanGestureHandler';
+import { longPressGestureHandlerProps } from '../LongPressGestureHandler';
+import { flingGestureHandlerProps } from '../FlingGestureHandler';
+import { forceTouchGestureHandlerProps } from '../ForceTouchGestureHandler';
+
+export abstract class SimpleGesture extends Gesture {
+  public handlerTag: number = -1;
+  public handlerName: string = '';
+  public config: any = {};
+  public handlers: any = {
+    handlerTag: -1,
+    onBegin: null,
+    onUpdate: null,
+    onEnd: null,
+    onStart: null,
+  };
+
+  //TODO fix type
+  static allowedProps: any = baseGestureHandlerWithMonitorProps;
+
+  constructor() {
+    super();
+  }
+
+  protected setConfig(key: string, value: any) {
+    this.config[key] = value;
+  }
+
+  protected setHandler(key: string, value: any) {
+    this.handlers[key] = value;
+  }
+
+  private addDependency(
+    key: string,
+    gesture: SimpleGesture | React.RefObject<SimpleGesture>
+  ) {
+    if (this.config[key]) {
+      if (Array.isArray(this.config[key])) {
+        this.config[key].push(gesture);
+      } else {
+        this.config[key] = [this.config[key], gesture];
+      }
```

Comment:
`this.config[key] = this.config[key] ? [].concat(this.config[key], gesture) : [gesture]`

It may be too dense so you can remove ternary if needed.

Reply:
I think it's fine, replaced in 23b5ab772057790132673b0b74507c0e86d131d0